pkg/controller/sync/controller.go

```diff
@@ -459,7 +459,7 @@ func (s *FederationSyncController) objectForUpdateOp(desiredObj, clusterObj *uns
 	if targetKind == util.ServiceAccountKind {
 		return serviceAccountForUpdateOp(desiredObj, clusterObj)
 	}
-	return desiredObj, nil
+	return retainReplicas(desiredObj, clusterObj, fedObj)
```

Comment:
Logically, the check on line 561 should happen here.
```
if retainReplicas {
    return retainReplicas(desiredObj, clusterObj, fedObj)
}

return desiredObj, nil
```
or change name to something like `retainReplicasIfNeeded` which actually sounds crude.

Reply:
Why is it problematic for the function to conditionally modify the resource?  Given that the test for retention is non-trivial, I would prefer not to cluster objectForUpdateOp with it.

Reply:
Readability; however, I don't see any harm in it being the way it is except the impression to the code reader here that all resources can have `replicas`.